packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -2,20 +2,190 @@
+///   problem that was detected.
+/// * [ErrorHint], which provides specific, non-obvious advice that may be
+///   applicable.
+/// * [FlutterError], which is the most common place to use an
+///   [ErrorDescription].
+class ErrorDescription extends _ErrorDiagnostic {
+  /// A lint enforces that this constructor can only be called with a string
+  /// literal to match the limitations of the Dart Kernel transformer that
+  /// optionally extracts out objects referenced using string interpolation in
+  /// the message passed in.
+  ///
+  /// The message will display with the same text regardless of whether the
+  /// kernel transformer is used. The kernel transformer is required so that
+  /// debugging tools can provide interactive displays of objects described by
+  /// the error.
+  ErrorDescription(String message) : super(message, level: DiagnosticLevel.info);
+
+  /// Calls to the default constructor may be rewritten to use this constructor
+  /// in debug mode using a kernel transformer.
+  ErrorDescription._fromParts(List<Object> messageParts) : super._fromParts(messageParts, level: DiagnosticLevel.info);
+}
+
+/// A short (one line) description of the problem that was detected.
+///
+/// Error summaries from the same source location should have little variance,
+/// so that they can be recognized as related. For example, they shouldn't
+/// include hash codes.
+///
+/// A [FlutterError] must start with an [ErrorSummary] and may not contain
+/// multiple summaries.
+///
+/// See also:
+///
+/// * [ErrorDescription], which provides an explanation of the problem and its
+///   cause, any information that may help track down the problem, background
+///   information, etc.
+/// * [ErrorHint], which provides specific, non-obvious advice that may be
+///   applicable.
+/// * [FlutterError], which is the most common place to use an [ErrorSummary].
+class ErrorSummary extends _ErrorDiagnostic {
+  /// A lint enforces that this constructor can only be called with a string
+  /// literal to match the limitations of the Dart Kernel transformer that
+  /// optionally extracts out objects referenced using string interpolation in
+  /// the message passed in.
+  ///
+  /// The message will display with the same text regardless of whether the
+  /// kernel transformer is used. The kernel transformer is required so that
+  /// debugging tools can provide interactive displays of objects described by
+  /// the error.
+  ErrorSummary(String message) : super(message, level: DiagnosticLevel.summary);
+
+  /// Calls to the default constructor may be rewritten to use this constructor
+  /// in debug mode using a kernel transformer.
+  ErrorSummary._fromParts(List<Object> messageParts) : super._fromParts(messageParts, level: DiagnosticLevel.summary);
+}
+
+/// An [ErrorHint] provides specific, non-obvious advice that may be applicable.
+///
+/// If your message provides obvious advice that is always applicable it is an
+/// [ErrorDescription] not a hint.
```

Comment:
If the advice is obvious from the summary/description of the error, it probably should not be included in the message? I feel this line is a bit extranous.

Reply:
I do like having the advice duplicated here as a reminder to only use hints for things that meet the fairly narrow charter of what a hint is.